File: LocalisationMateriel/src/java/metier/BatimentServiceImpl.java
package metier;

import java.util.ArrayList;
import java.util.List;
import physique.BatimentDataService;
import physique.PhysiqueFactory;

public class BatimentServiceImpl implements BatimentService {

    private final BatimentDataService batimentDataSrv;

    BatimentServiceImpl() {
        batimentDataSrv = PhysiqueFactory.getBatimentDataService();
    }

    @Override
    public Batiment add(Batiment batiment) {
        Batiment bat = null;
        try {
            if (!batiment.getNomBatiment().isEmpty() && !batiment.getPlan().isEmpty()) {
                bat = batimentDataSrv.add(batiment);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return bat;
    }

    @Override
    public boolean remove(Batiment batiment) {
        boolean execute = false;
        try {
            if (batiment.getId() > 0) {
                execute = batimentDataSrv.remove(batiment);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return execute;
    }

    @Override
    public boolean update(Batiment batiment) {
        boolean execute = false;
        try {
            if (batiment.getId() > 0) {
                if (!batiment.getNomBatiment().isEmpty() && !batiment.getPlan().isEmpty()) {
                    execute = batimentDataSrv.update(batiment);
                }
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return execute;
    }

    @Override
    public List<Batiment> getAll(int debut, int parPage) {
        List<Batiment> bat = new ArrayList<Batiment>();
        try {
            bat = batimentDataSrv.getAll(debut, parPage);
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return bat;
    }

    @Override
    public int getCount() {
        int count = 0;
        try {
            count = batimentDataSrv.getCount();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

    @Override
    public Batiment getById(long id) {
        Batiment bat = null;
        try {
            if (id > 0) {
                bat = batimentDataSrv.getById(id);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return bat;
    }

    @Override
    public Batiment getByNomBatiment(String nomBatiment) {
        Batiment bat = null;
        try {
            if (!nomBatiment.isEmpty()) {
                bat = batimentDataSrv.getByNomBatiment(nomBatiment);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return bat;
    }

    @Override
    public List<String> getListNomBatiment(int debut, int parPage) {
        List<String> listStr = new ArrayList<String>();
        try {
            if (debut >= 0 && parPage > 0) {
                listStr = batimentDataSrv.getListNomBatiment(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return listStr;
    }

    @Override
    public int getCountListNomBatiment() {
        int count = 0;
        try {
            count = batimentDataSrv.getCountListNomBatiment();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

    @Override
    public Batiment getBySalle(Salle salle) {
        Batiment bat = null;
        try {
            if (salle.getId() > 0) {
                bat = batimentDataSrv.getBySalle(salle);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return bat;
    }

    @Override
    public List<Salle> getListSalle(int debut, int parPage) {
        List<Salle> salles = new ArrayList<Salle>();
        try {
            if (debut >= 0 && parPage > 0) {
                salles = batimentDataSrv.getListSalle(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return salles;
    }

    @Override
    public int getCountListSalle() {
        int count = 0;
        try {
            count = batimentDataSrv.getCountListSalle();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

    @Override
    public Batiment getByPlan(String plan) {
        Batiment bat = null;
        try {
            if (!plan.isEmpty()) {
                bat = batimentDataSrv.getByPlan(plan);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return bat;
    }

    @Override
    public List<String> getListPlan(int debut, int parPage) {
        List<String> listStr = new ArrayList<String>();
        try {
            if (debut >= 0 && parPage > 0) {
                listStr = batimentDataSrv.getListPlan(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return listStr;
    }

    @Override
    public int getCountListPlan() {
        int count = 0;
        try {
            count = batimentDataSrv.getCountListPlan();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

}

File: AppMateriel/metier/batimentserviceimpl.cpp
#include "batimentserviceimpl.h"

Metier::BatimentServiceImpl::BatimentServiceImpl()
{
    batimentDataSrv = PhysiqueFactory::getBatimentDataServiceImpl();
}

Batiment Metier::BatimentServiceImpl::add(Batiment batiment)
{
    Batiment bat;
    if(!batiment.getNomBatiment().isEmpty() && !batiment.getPlan().isEmpty()){
        bat = batimentDataSrv->add(batiment);
    }
    return bat;
}

bool Metier::BatimentServiceImpl::remove(Batiment batiment)
{
    bool execute = false;
    if(batiment.getId() > 0){
        execute = batimentDataSrv->remove(batiment);
    }
    return execute;
}

bool Metier::BatimentServiceImpl::update(Batiment batiment)
{
    bool execute = false;
    if(batiment.getId() > 0){
        if(!batiment.getNomBatiment().isEmpty() && !batiment.getPlan().isEmpty()){
            execute = batimentDataSrv->update(batiment);
        }
    }
    return execute;
}

QList<Batiment> Metier::BatimentServiceImpl::getAll(int debut, int parPage)
{
    QList<Batiment> bat;
    if(debut >= 0 && parPage > 0){
        bat = batimentDataSrv->getAll(debut, parPage);
    }
    return bat;
}

int Metier::BatimentServiceImpl::getCount()
{
    return batimentDataSrv->getCount();
}

Batiment Metier::BatimentServiceImpl::getById(long id)
{
    Batiment bat;
    if(id > 0){
        bat = batimentDataSrv->getById(id);
    }
    return bat;
}

Batiment Metier::BatimentServiceImpl::getByNomBatiment(QString nomBatiment)
{
    Batiment bat;
    if(!nomBatiment.isEmpty()){
        bat = batimentDataSrv->getByNomBatiment(nomBatiment);
    }
    return bat;
}

QList<QString> Metier::BatimentServiceImpl::getListNomBatiment(int debut, int parPage)
{
    QList<QString> listStr;
    if(debut >= 0 && parPage > 0){
        listStr = batimentDataSrv->getListNomBatiment(debut, parPage);
    }
    return listStr;
}

int Metier::BatimentServiceImpl::getCountListNomBatiment()
{
    int count = batimentDataSrv->getCountListNomBatiment();
    return count;
}

Batiment Metier::BatimentServiceImpl::getBySalle(Salle salle)
{
    Batiment bat;
    if(salle.getId() > 0){
        bat = batimentDataSrv->getBySalle(salle);
    }
    return bat;
}

QList<Salle> Metier::BatimentServiceImpl::getListSalle(int debut, int parPage)
{
    QList<Salle> salles;
    if(debut >= 0 && parPage > 0){
        salles = batimentDataSrv->getListSalle(debut, parPage);
    }
    return salles;
}

int Metier::BatimentServiceImpl::getCountListSalle()
{
    int count = batimentDataSrv->getCountListSalle();
    return count;
}

Batiment Metier::BatimentServiceImpl::getByPlan(QString plan)
{
    Batiment bat;
    if(!plan.isEmpty()){
        bat = batimentDataSrv->getByPlan(plan);
    }
    return bat;
}

QList<QString> Metier::BatimentServiceImpl::getListPlan(int debut, int parPage)
{
    QList<QString> listStr;
    if(debut >= 0 && parPage > 0){
        listStr = batimentDataSrv->getListPlan(debut, parPage);
    }
    return listStr;
}

int Metier::BatimentServiceImpl::getCountListPlan()
{
    int count = batimentDataSrv->getCountListPlan();
    return count;
}
